### PR TITLE
Add broken implementation note for DLS and UPnP

### PR DESCRIPTION
**Pull request by catree**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [X] The PR is proposed to proper branch
- [X] There is reference to original bug report and related work
- N/A There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- N/A The feature is well documented and sample code can be built with the project CMake

This is 5 years too late in my opinion.

See #5065 and https://github.com/opencv/opencv/pull/6864#issuecomment-473450679

### CODE REVIEW

File: modules/calib3d/include/opencv2/calib3d.hpp [hunk 1]
```diff
@@ -450,8 +450,10 @@ enum SolvePnPMethod {
     SOLVEPNP_ITERATIVE   = 0,
     SOLVEPNP_EPNP        = 1, //!< EPnP: Efficient Perspective-n-Point Camera Pose Estimation @cite lepetit2009epnp
     SOLVEPNP_P3P         = 2, //!< Complete Solution Classification for the Perspective-Three-Point Problem @cite gao2003complete
-    SOLVEPNP_DLS         = 3, //!< A Direct Least-Squares (DLS) Method for PnP  @cite hesch2011direct
-    SOLVEPNP_UPNP        = 4, //!< Exhaustive Linearization for Robust Camera Pose and Focal Length Estimation @cite penate2013exhaustive
+    SOLVEPNP_DLS         = 3, //!< <b> Broken implementation. Using this flag will fallback to EPnP. </b>
+                              //!< <del> A Direct Least-Squares (DLS) Method for PnP @cite hesch2011direct </del>
```
Comment (alalek) [inline modules/calib3d/include/opencv2/calib3d.hpp:454]: > `calib3d/include/opencv2/calib3d.hpp:454: warning: Unsupported xml/html tag <del> found`

http://pullrequest.opencv.org/buildbot/export/pr/18171/docs/d9/d0c/group__calib3d.html#ga357634492a94efe8858d0ce1509da869

Can we use `@note` here instead?

Related SO question: https://stackoverflow.com/questions/27827966/how-can-i-use-strikethrough-in-the-doxygens-markdown

Comment (catree) [reply modules/calib3d/include/opencv2/calib3d.hpp]: I have replaced `<del>` with `~~`. Let see if it works.

Comment (catree) [reply modules/calib3d/include/opencv2/calib3d.hpp]: Looks like strikethrough is not correctly rendered with the version of Doxygen on the CI. I will update to remove it.

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**